extensions/ql-vscode/src/extension.ts

```diff
@@ -782,12 +770,13 @@ async function activateWithInstalledDistribution(
           });
         }
 
-        if (queryUris.length > 1) {
+        if (queryUris.length > 1 && !await cliServer.cliConstraints.supportsNonDestructiveUpgrades()) {
```

Comment:
Interesting...we never had this before. Probably a bug that we always tried to upgrade the database. Most likely in these cases, this was a no-op that just took some time.